code/scripting/api/libs/ui.cpp

```diff
@@ -885,40 +885,57 @@ ADE_FUNC(canSkip,
 
 ADE_FUNC(replayMission,
 	l_UserInterface_Debrief,
+	"[boolean restart = false]",
+	"Resets the mission outcome, and optionally restarts the mission at the briefing; "
+	"true to restart the mission, false to stay at current UI. Defaults to true.",
 	nullptr,
-	"Resets the mission outcome and begins the current mission again",
-	"number",
-	"Returns true when completed")
+	nullptr)
 {
+
+	bool restart = true;
+
+	ade_get_args(L, "|b", &restart);
+
 	// This is used to skip some UI preloading in debrief init
 	API_Access = true;
 
 	debrief_close();
 
-	gameseq_post_event(GS_EVENT_START_GAME);
+	if (restart) {
+		gameseq_post_event(GS_EVENT_START_GAME);
+	}
 
 	API_Access = false;
 
-	return ade_set_args(L, "b", true);
+	return ADE_RETURN_NIL;
```

Comment:
Is it okay to change the return type? A previous script that used this would possibly break, but I'm not sure if it's really something that needs to be checked (in which case other scripts probably never used it)

Reply:
Unless someone wrote a script in the last day other than me, then yes, it's OK to change.